go-controller/pkg/cni/cnishim.go

```diff
@@ -176,6 +178,29 @@ func (c *shimClientset) getPod(namespace, name string) (*kapi.Pod, error) {
 	return c.kclient.CoreV1().Pods(namespace).Get(context.TODO(), name, metav1.GetOptions{})
 }
 
+func initPodRequestUnprivilegedMode(req *Request, isDPUHostMode bool) (*PodRequest, error) {
+	var vsClient libovsdbclient.Client
+
+	if !isDPUHostMode {
+		// Initialize OVS exec runner; find binaries that the CNI code uses.
+		if err := SetExec(kexec.New()); err != nil {
+			return nil, fmt.Errorf("failed to initialize OVS exec runner: %w", err)
+		}
+	}
+
+	vsClient, err := libovsdb.NewVSwitchClient(make(chan struct{}))
```

Comment:
Surprised that we get to use the client in the shim as well, and and interesting performance comparison here, I guess it is still a win.

Reply:
@jcaamano unprivileged mode does all the OVS stuff and netlink/namespace stuff from the shim, so we kind of have to do it here too. It's a mode that NVIDIA added and uses so they don't run a bunch of heavy/privileged operations from the container itself. It's not a bad thing, *except* you don't have a long-running process with caches and stuff that you can read when waiting for the pod setup; and you have to prime the cache every time the shim starts. So unprivileged mode should have worse performance actually.